api/datastore/datastore.go

```diff
@@ -13,12 +13,7 @@ import (
 )
 
 func New(ctx context.Context, dbURL string) (models.Datastore, error) {
-	ds, err := newds(ctx, dbURL) // teehee
-	if err != nil {
-		return nil, err
-	}
-
-	return Wrap(ds), nil
+	return newds(ctx, dbURL) // teehee
 }
 
 func Wrap(ds models.Datastore) models.Datastore {
```

Comment:
Two thoughts on this:
 * Does the system ever want to create a non-metric creating datastore? If not then this should be moved in 'newds' so it can't be forgotten.
* Wrap is very generic. DecorateMetrics, or even AddMetrics would help readability, for me at least.

Reply:
CP may rely on this Wrap() being inside New() and in WithDatastore()

Reply:
hmm, confused by the wording there. if a caller uses `WithDatastore` with their own datastore, then the behavior here should remain unchanged - that is, the provided datastore will get wrapped [once], just in New() instead of WithDatastore().